src/Cli/dotnet/SlnFileFactory.cs

```diff
@@ -19,87 +49,57 @@ public static string[] ListSolutionFilesInDirectory(string directory, bool inclu
-                        solutionDirectory);
-                }
-            }
-            catch (ArgumentException)
-            {
-                throw new GracefulException(
-                    CommonLocalizableStrings.CouldNotFindSolutionOrDirectory,
-                    solutionDirectory);
-            }
-
-            FileInfo[] files = [
-                ..dir.GetFiles("*.sln"),
-                ..(includeSolutionXmlFiles ? dir.GetFiles(".slnx") : [])
-             ];
+            JsonDocument jsonDocument = JsonDocument.Parse(File.ReadAllText(filteredSolutionFile));
+            JsonElement jsonElement = jsonDocument.RootElement;
+            JsonElement filteredSolutionJsonElement = jsonElement.GetProperty("solution");
+            string originalSolutionPath = filteredSolutionJsonElement.GetProperty("path").GetString();
+            string originalSolutionPathAbsolute = Path.GetFullPath(Path.GetDirectoryName(originalSolutionPath), filteredSolutionFile);
+            string[] filteredSolutionProjectPaths = filteredSolutionJsonElement.GetProperty("projects")
+                .EnumerateArray()
+                .Select(project => project.GetString())
+                .ToArray();
 
-            if (files.Length == 0)
+            if (string.IsNullOrEmpty(originalSolutionPathAbsolute) || !File.Exists(originalSolutionPathAbsolute))
             {
                 throw new GracefulException(
-                    CommonLocalizableStrings.CouldNotFindSolutionIn,
-                    solutionDirectory);
+                    CommonLocalizableStrings.InvalidSolutionFormatString,
+                    filteredSolutionFile, "");
             }
 
-            if (files.Length > 1)
-            {
-                throw new GracefulException(
-                    CommonLocalizableStrings.MoreThanOneSolutionInDirectory,
-                    solutionDirectory);
-            }
+            SolutionModel filteredSolution = new SolutionModel();
+            SolutionModel originalSolution = CreateFromFileOrDirectory(originalSolutionPath);
 
-            FileInfo solutionFile = files.Single();
-            if (!solutionFile.Exists)
+            foreach (string path in filteredSolutionProjectPaths)
             {
-                throw new GracefulException(
-                    CommonLocalizableStrings.CouldNotFindSolutionIn,
-                    solutionDirectory);
+                SolutionProjectModel? project = originalSolution.FindProject(path);
+                if (project is null)
+                {
+                    throw new GracefulException(
+                        CommonLocalizableStrings.ProjectNotFoundInTheSolution,
+                        path,
+                        originalSolutionPath);
+                }
+                filteredSolution.AddProject(project.FilePath, project.Type, project.Parent is null ? null : filteredSolution.AddFolder(project.Parent.Path));
```

Comment:
Is there anything else that needs to be transferred over in addition to some projects?

Reply:
Only other thing I can think of is build types / configurations, and platforms

Reply:
Random (sort of related) question. Have these new solution commands been tested on solutions without .NET projects in them? Since solutions are not .NET specific, they could just be a hodgepodge of different projects. But our solution CLI commands (I would think) should be able to operate on solutions with any kind of project in them.